File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/usecases/photos/UpdatePhotoPropertyUseCase.java
package nl.acidcats.tumblrlikes.core.usecases.photos;

import nl.acidcats.tumblrlikes.core.models.Photo;
import rx.Observable;

/**
 * Created on 03/10/2018.
 */
public interface UpdatePhotoPropertyUseCase {
    Observable<Photo> updateLike(long id, boolean isLiked);

    Observable<Photo> updateFavorite(long id, boolean isFavorite);

    Observable<Photo> setHidden(long id);
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/core/usecases/photos/UpdatePhotoPropertyUseCase.kt
package nl.acidcats.tumblrlikes.core.usecases.photos

import nl.acidcats.tumblrlikes.core.models.Photo
import nl.acidcats.tumblrlikes.core.viewmodels.PhotoViewModel
import rx.Observable

/**
 * Created on 03/10/2018.
 */
interface UpdatePhotoPropertyUseCase {
    fun updateLike(id: Long, isLiked: Boolean): Observable<PhotoViewModel>

    fun updateFavorite(id: Long, isFavorite: Boolean): Observable<PhotoViewModel>

    fun setHidden(id: Long): Observable<PhotoViewModel>
}
